appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/eesecurity/SignedJWTIdentityStore.java

```diff
@@ -211,10 +214,10 @@ public CredentialValidationResult validate(SignedJWTCredential signedJWTCredenti
             return Optional.empty();
         }
 
-        byte[] byteBuffer = new byte[16384];
-        try (InputStream inputStream = publicKeyURL.openStream()) {
-            String key = new String(byteBuffer, 0, inputStream.read(byteBuffer));
-            return createPublicKey(key, keyID);
+        try (InputStream inputStream = publicKeyURL.openStream();
+             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))){
```

Comment:
Shouldn't you explicitly use a CharSet to read from the stream (using the InputStreamReader ctor with CharSet) instead of using the system default CharSet?

Reply:
Thanks for the suggestion